packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ArtifactController.java

```diff
@@ -70,6 +72,7 @@ public ResponseEntity<List<Artifact>> getArtifacts(
 	@PostMapping
 	@Secured(Roles.USER)
 	public ResponseEntity<JsonNode> createArtifact(@RequestBody Artifact artifact) throws IOException {
+		artifact.setCreatedOn(Timestamp.from(Instant.now()));
```

Comment:
Does it make sense to encapsulate this in the `createXXX`, `updateXXX`, and `deleteXXX` methods instead of the controller?